File: BookShop/ClientApp/src/redux/types/book/book.ts
export enum fetchBooksActions {
    FETCH_BOOKS = 'FETCH_BOOKS',
    FETCH_BOOKS_SUCCESS = 'FETCH_BOOKS_SUCCESS',
    FETCH_BOOKS_ERROR = 'FETCH_BOOKS_ERROR'
};

export type book = {
    name: string;
    author: string;
    description: string;
    price: number;
    pages: number;
    imageURL: string;
    genre: string
    edition: string;
    id: number;
}

export interface booksState {
    books: fetchedBookType[];
    loading: boolean;
    error: null | string;
}

export type fetchBookAction = bookFetchAction | bookFetchSuccessAction | bookFetchErrorAction

interface bookFetchAction {
    type: fetchBooksActions.FETCH_BOOKS;
 }

interface bookFetchSuccessAction {
    type: fetchBooksActions.FETCH_BOOKS_SUCCESS;
    payload: fetchedBookType[];
}

interface bookFetchErrorAction {
    type: fetchBooksActions.FETCH_BOOKS_ERROR;
    payload: string
}

export type fetchBooksOptions = {
    default: boolean;
    name?: string;
    priceAbove?: number;
    priceBelow?: number;
    genre?: string;
}

export type bookInput = {
    name: string;
    description: string;
    price: number;
    pages: number;
    imageURL: string;
    edition: string;
    id: number;
}

export type bookInputParams = {
    author: string[];
    genre: string[];
    name: string;
}

export type fetchedBookType = {
    name: string;
    author: string[];
    description: string;
    hasInStorage: boolean;
    price: number;
    pages: number;
    imageURL: string;
    genre: string[];
    edition: string;
    id: number;
}

export type addBookInStorageType = {
    name: string;
    count: number;
}

File: BookShop/Models/BooksParams/Book.cs
using System.Collections.Generic;

namespace BookShop.Models
{
    public class Book
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string ImageURL { get; set; }
        public string Edition { get; set; }
        public int Pages { get; set; }
        public int Price { get; set; }
        public bool HasInStorage { get; set; }
    }
    public class InputBookData
    {
        public string Name { get; set; }
        public List<string> Author { get; set; }
        public List<string> Genre { get; set; }
    }
    
    public class ClientBook
    {
        public ClientBook(Book book)
        {
            this.Description = book.Description;
            this.Edition = book.Edition;
            this.Name = book.Name;
            this.Pages = book.Pages;
            this.ImageURL = book.ImageURL;
            this.Price = book.Price;
            this.Id = book.Id;
        }
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string ImageURL { get; set; }
        public string Edition { get; set; }
        public int Pages { get; set; }
        public int Price { get; set; }
        public bool HasInStorage { get; set; }
        public List<string> Author { get; set; }
        public List<string> Genre { get; set; }
    }
}
